src/nix/doctor.cc

```diff
@@ -130,6 +135,15 @@ struct CmdDoctor : StoreCommand
 
         return checkPass("Client protocol matches store protocol.");
     }
+
+    void checkTrustedUser(ref<Store> store)
+    {
+        bool trusted = store->checkTrustedUser();
```

Comment:
This will throw an error if the remote store is too old, possibly prematurely terminating the doctor, which feels a little surprising. I wouldn't call it a dealbreaker though.

Reply:
It doesn't change the exit status, and I think it's the correct way to fail. Failure is not a form of `[INFO]`, only the successful case of figuring out if we're trusted or not is. Anything else should just be printed as-is, which is how it works at the moment.

Reply:
Apparently you need to bit shift by 8 bytes, this was incredibly confusing to me because:
- `GET_PROTOCOL_MAJOR(conn->daemonVersion)` will return `256`
- `GET_PROTOCOL_MAJOR(conn->daemonVersion) >> 8` will return `1`

Anyway.. I've implemented it now as follows.

```diff
diff --git a/src/libstore/remote-store.cc b/src/libstore/remote-store.cc
index 154afa726..5cfd432dd 100644
--- a/src/libstore/remote-store.cc
+++ b/src/libstore/remote-store.cc
@@ -1092,7 +1092,8 @@ bool RemoteStore::checkTrustedUser()
         conn.processStderr();
         return readInt(conn->from);
     } else {
-      throw Error("daemon protocol %d is too old (< 2.13) to ask whether we're a trusted user",
+      throw Error("daemon protocol %d.%d is too old (< 2.13) to ask whether we're a trusted user",
+          GET_PROTOCOL_MAJOR(conn->daemonVersion) >> 8,
           GET_PROTOCOL_MINOR(conn->daemonVersion));
     }
 }
```